layer/memory_usage_layer.cc

```diff
@@ -26,13 +34,47 @@ constexpr char kLayerName[] = "VK_LAYER_STADIA_memory_usage";
 constexpr char kLayerDescription[] = "Stadia Memory Usage Measuring Layer";
 constexpr char kLogFilenameEnvVar[] = "VK_MEMORY_USAGE_LOG";
 
-class MemoryUsageLayerData : public performancelayers::LayerData {
+// An event that holds memory allocation information (current and peak
+// allocated) and can be logged both in the private and common files.
+class MemoryUsageEvent : public Event {
+ public:
+  MemoryUsageEvent(const char* name, int64_t current, int64_t peak)
+      : Event(name, LogLevel::kHigh),
+        current_({"current", current}),
+        peak_({"peak", peak}),
+        attributes_{&current_, &peak_} {}
+
+  const std::vector<Attribute*>& GetAttributes() override {
+    return attributes_;
+  }
+
+  size_t GetNumAttributes() const override { return attributes_.size(); }
```

Comment:
Can we make this the default implementation for all `Event`s? I expect all of them to implement it the same way anyway.

Reply:
Yes. Sure.